### DISCUSSION TITLE
Creating a custom renderer for an image and other content

### DISCUSSION THREAD

**Discussion by pkonda2019** [remarkjs/remark#1038] [Q&A] [answered]
```
const markdownSource = `
![Image](https://www.iheartradio.ca/image/policy:1.15731844:1627581512/rick.jpg?f=default&$p$f=20c1bb3)Exclusive **Aloe Vera Bio Ferment** helps skin retain moisture. Created using responsibly sourced and Certified Organic **Aloe Extract Powder, Lactobacillus** and **Activated Aloe Water**.

`;
```
From above markdownSource I want image should be render in a separate `<div>` and remaining entire text content should be render in another single `<div>`.
Using an example from @ChristianMurphy ([https://codesandbox.io/s/react-markdown-debug-forked-6789z](https://codesandbox.io/s/react-markdown-debug-forked-6789z)) I can add `<div>` element to Image (shown in below sample) but not able to add the same to entire remaining text content. Below is the sample:

```
<Markdown
      components={{
        img: (props) => **<div>**<img {...props} style={{ maxWidth: "80px" }} alt="" />**</div>**
      }}
    >
      {markdownSource}
</Markdown>
```

Expected output below:

```
<div><img src="https://www.iheartradio.ca/image/policy:1.15731844:1627581512/rick.jpg?f=default&$p$f=20c1bb3"></div>
<div>Exclusive <strong>Aloe Vera Bio Ferment</strong> helps skin retain moisture. Created using responsibly sourced and Certified Organic <strong>Aloe Extract Powder, Lactobacillus</strong> and <strong>Activated Aloe Water</strong>.</div>
Thanks in advance.
```

**Comment by ChristianMurphy** [comment, marked as answer]
Plain commonmark/gfm really don't offer a way to group things together in the way you're describing.
You could try writing an AST plugin to find content that matches this pattern and transform it into the structure you want https://unifiedjs.com/learn/

---

That said, it sounds like you're looking for either something like more like MDX or directives.

MDX source for a figure would look something like:

```jsx
<Figure imageSrc="https://www.iheartradio.ca/image/policy:1.15731844:1627581512/rick.jpg?f=default&$p$f=20c1bb3">
  Exclusive **Aloe Vera Bio Ferment** helps skin retain moisture. Created using responsibly sourced and Certified Organic **Aloe Extract Powder, Lactobacillus** and **Activated Aloe Water**.
</Figure>
```

You can learn more on MDX here: https://mdxjs.com/

---

Markdown + Directives source for a figure would look something like:

```markdown
:::figure{imgSrc="https://www.iheartradio.ca/image/policy:1.15731844:1627581512/rick.jpg?f=default&$p$f=20c1bb3" imgAlt="Image"}

Exclusive **Aloe Vera Bio Ferment** helps skin retain moisture. Created using responsibly sourced and Certified Organic **Aloe Extract Powder, Lactobacillus** and **Activated Aloe Water**.

:::
```

More on remark-directive can be found here: https://github.com/remarkjs/remark-directive and https://github.com/orgs/remarkjs/discussions/625
